src/app/view/transport-calls-table/transport-calls-table.component.ts

```diff
@@ -72,47 +55,63 @@ export class TransportCallsTableComponent implements OnInit {
     return transportCall.omitCreatedDateTime >= transportCall.latestEventCreatedDateTime;
   }
 
-  async refreshTransportCalls(): Promise<void> {
-    this.progressing = true;
-    const transportCalls = await this.loadTransportCalls();
-    if (this.selectedTransportCall && !transportCalls.some(x => x.transportCallID === this.selectedTransportCall.transportCallID)) {
-      this.transportCallNotifier.emit(null);
-    }
+  refreshTransportCalls(): void {
+    this.refreshTrigger.next(null);
   }
 
-  openCreationDialog() {
+  openCreationDialog(): void {
     const transportCallEditor = this.dialogService.open(TransportCallCreatorComponent, {
       header: this.translate.instant('general.portVisit.create'),
       width: '75%'
     });
-    transportCallEditor.onClose.subscribe(async result => {
+    transportCallEditor.onClose.pipe(take(1)).subscribe(result => {
       if (result) {
-        await this.refreshTransportCalls();
+        this.refreshTransportCalls();
       }
-    })
+    });
   }
 
-  async loadTransportCalls(): Promise<TransportCall[]> {
-    return new Promise(resolve => {
-      this.transportCallService.getTransportCalls(this.filterPort?.UNLocationCode, this.filterVessel?.vesselIMONumber).subscribe({
+  fetchTransportCalls(): Observable<TransportCall[]> {
+    // The getPortsCall ensures all ports are cached.
+    return this.portService.getPorts().pipe(
+      mergeMap(_ => {
+        return combineLatest([
+          this.portFilterService.portObservable$,
+          this.portFilterService.vesselObservable$,
+          this.vesselService.vesselsObservable$,
+          this.refreshTrigger,
+        ]);
+      }),
+      mergeMap(([
+                  filterPort,
+                  filterVessel,
+                  _a,  // Unused (used as a reload-trigger-only to refresh vessel information)
+                  _b,  // Unused (used as a reload-trigger-only - manual refresh request)
+                ]) => {
+        return this.transportCallService.getTransportCalls(filterPort?.UNLocationCode, filterVessel?.vesselIMONumber);
+      }),
+      tap({
         next: (transportCalls) => {
-          this.progressing = false;
-          this.transportCalls = transportCalls;
-          resolve(transportCalls);
+          if (this.selectedTransportCall) {
+            const id = this.selectedTransportCall.transportCallID;
```

Comment:
This also means that we stay selected on the same transport call, even though it is not included on the list of shown TCs. 

Assuming we selected a TC and filter on port where _filtered port_ != _selected TC port_ we still stuck selecting it.

Reply:
That is not a new problem as far as I can tell.  What is new is that we reload the existing selection with the latest information.

Reply:
Yes, it is not I misread. The issue is on the timestamp-table side since we do not handle a selected TC being null correctly.